### DISCUSSION TITLE
Reduce CPU load

### DISCUSSION THREAD

**Discussion by addlistener** [trpc/trpc#3097] [Ideas]
I love TRPC. However I can feel type hints lagging and cpu roaring when apply it in a bigger project. Is there something we can do to reduce the type calculation?

**Comment by cqh963852** [comment]
We can divide the problem into a little more detail. 
- For the frontend.
- For the backend.

---
For the backend

I like input validation on the backend. The type of routing request parameters is generated by schema

So currently type inference is `schema => type`, which I think is perfect

---

For the frontend

1. The type of useQuery is generated from the schema. I don't think it's necessary.

You can check my problem on https://github.com/omar-dulaimi/prisma-trpc-generator/discussions/61

Although I generated the schema required by the backend. The frontend cannot correctly infer the parameters I need, and I still need to manually modify

My idea is to make some tweaks on the type of useQuery .

![image](https://user-images.githubusercontent.com/17702287/213872081-b36e6e70-f349-4618-a02a-892f6d00bc0d.png)

The first generic parameter is the return value type.

The second generic parameter is the type of query.

That would be easier to use.

**Comment by cqh963852** [comment]
For the backend

While checking the router it will show me  `ctx` `meta`, which is not my concern.

I'm more concerned with what fields my routes have. What type should the route return for each field.

![image](https://user-images.githubusercontent.com/17702287/213872274-49dc91fa-4458-42ed-8042-855ab234acac.png)

**Comment by addlistener** [comment]
My 2 cents

Is it possible we do it by directly referencing `function backEndApi()` in frontend and apply transformations in babel to generate the actual server code?
